### PR TITLE
Bugfix: Solana linting

### PR DESCRIPTION
**Pull request by lambertkevin**
## 🦒 Context (issues, jira)

The integration of Solana has some linting issues. This PR should fix the errors.

## 🖤  Expectations to reach

PR must pass CI, rebase develop if conflicts. Thanks!

- **on QA**: at least one of these two checkboxes must be checked:
  - [ ] a specific test planned is defined on Jira
  - [ ] this PR is covered by automatic UI test
- **on delivery**: at least one of these two checkboxes must be checked: 
  - [x] Option 1: **no impact**: The changes of this PR have ZERO impact on the userland (invisible for users)
  - [ ] Option 2: **atomic delivery**: the changes is atomic and complete (no partial delivery)

### CODE REVIEW

File: src/renderer/families/solana/shared/fields/ValidatorsField.js [hunk 1]
```diff
@@ -28,13 +25,9 @@ type Props = {
...
 
 const ValidatorField = ({ t, account, onChangeValidator, chosenVoteAccAddr, status }: Props) => {
-  if (!status) return null;
-
   invariant(account && account.solanaResources, "solana account and resources required");
 
-  const { solanaResources } = account;
-
-  const [search, setSearch] = useState("");
+  const [search] = useState("");
```
Comment (lambertkevin) [inline src/renderer/families/solana/shared/fields/ValidatorsField.js:36]: Removed the setter for search, which is weird. Is it not weird that it's not used ?

File: src/renderer/families/solana/shared/fields/ValidatorsField.js [hunk 2]
```diff
@@ -27,14 +24,16 @@ type Props = {
   onChangeValidator: (v: ValidatorAppValidator) => void,
 };
 
-const ValidatorField = ({ t, account, onChangeValidator, chosenVoteAccAddr, status }: Props) => {
+const MaybeValidatorField = ({ status }: Props) => {
   if (!status) return null;
 
-  invariant(account && account.solanaResources, "solana account and resources required");
+  return ValidatorField;
+};
```
Comment (lambertkevin) [inline src/renderer/families/solana/shared/fields/ValidatorsField.js:31]: This is here to replace the `if (!status) return null;` cause conditional hooks are not allowed by React.

**Review by lambertkevin — COMMENTED** [review]
Fixed the linting + an optional component.

Still needs to be tested cause I removed a lot of "unused" code and I don't know if it's normal or not.

File: src/renderer/families/solana/shared/fields/ValidatorsField.js (hunk 1 above)
Comment (haammar-ledger) [reply src/renderer/families/solana/shared/fields/ValidatorsField.js:36]: Probably either some forgotten or abandoned design. No one complained so :+1: to remove.

**Review by ghost — APPROVED** [review]
Code LGTM. Thanks a lot!
Why no CI though

**Review by ghost — APPROVED** [review]
All good at last, just need to target `release/2.41.x` instead of `develop`.

**Pull request merged**